aws-java-sdk-dynamodb/src/test/java/com/amazonaws/services/dynamodbv2/document/utils/ValueMapAndWithJsonSupportTest.java

```diff
@@ -0,0 +1,44 @@
+package com.amazonaws.services.dynamodbv2.document.utils;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.*;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.amazonaws.AmazonClientException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+public class ValueMapAndWithJsonSupportTest {
+
+    private static final String NO_JSON_STRING = "nojson";
+    private static final String KEY = "somekey";
+
+    @Test(expected = AmazonClientException.class)
+    public void valueMapCreationShouldFailIfNoJsonStringIsUsedAsValue() {
+        new ValueMap().withJSON("a", NO_JSON_STRING);
+    }
+
+    @Test
+    public void valueMapShouldReturnAProperDeserializedJsonObject() {
+        Map<String, Object> expected = new LinkedHashMap<String, Object>();
+        expected.put("fruit", "pair");
+        expected.put("color", "green");
```

Comment:
I get that mapper.writeValueAsString is going to convert this to `{"fruit": "pair", "color": "green" }` but since this feature is to convert a string (and avoid having to do the conversion to an object) it might make sense for this test to show-case the _actual_ input string. So when I'm reading the test I can look at the actual JSON string above.